waku/v2/protocol/waku_rln_relay/rln_relay.nim

```diff
@@ -0,0 +1,418 @@
+  var credentials: MembershipCredentials
+  var persistCredentials = false
+  # create an RLN instance
+  let rlnInstanceRes = createRLNInstance()
+  if rlnInstanceRes.isErr():
+    raise newException(CatchableError, "RLN instance creation failed")
+  let rlnInstance = rlnInstanceRes.get()
+  if not conf.rlnRelayDynamic:
+    # static setup
+    let parsedGroupKeysRes = StaticGroupKeys.toIdentityCredentials()
+    if parsedGroupKeysRes.isErr():
+      raise newException(ValueError, "Static group keys are not valid")
+    groupManager = StaticGroupManager(groupSize: StaticGroupSize,
+                                      groupKeys: parsedGroupKeysRes.get(),
+                                      membershipIndex: conf.rlnRelayMembershipIndex,
+                                      rlnInstance: rlnInstance)
+    # we don't persist credentials in static mode since they exist in ./constants.nim
+    # TODO: registration handler
+  else:
+    # dynamic setup
+    let ethPrivateKey = if conf.rlnRelayEthAccountPrivateKey == "": none(string) else: some(conf.rlnRelayEthAccountPrivateKey)
+    let rlnRelayCredPath = if conf.rlnRelayCredPath == "": none(string) else: some(conf.rlnRelayCredPath)
+    let rlnRelayCredentialsPassword = if conf.rlnRelayCredentialsPassword == "": none(string) else: some(conf.rlnRelayCredentialsPassword)
+    groupManager = OnchainGroupManager(ethClientUrl: conf.rlnRelayEthClientAddress,
+                                       ethContractAddress: $conf.rlnRelayEthContractAddress,
+                                       ethPrivateKey: ethPrivateKey,
+                                       rlnInstance: rlnInstance,
+                                       registrationHandler: registrationHandler,
+                                       keystorePath: rlnRelayCredPath,
+                                       keystorePassword: rlnRelayCredentialsPassword,
+                                       saveKeystore: true)
+
+  # Initialize the groupManager
+  await groupManager.init()
+  # Start the group sync
+  await groupManager.startGroupSync()
+
+  return WakuRLNRelay(pubsubTopic: conf.rlnRelayPubsubTopic,
+                      contentTopic: conf.rlnRelayContentTopic,
+                      groupManager: groupManager)
+
+
+proc new*(T: type WakuRlnRelay,
+          conf: WakuRlnConfig,
+          registrationHandler: Option[RegistrationHandler] = none(RegistrationHandler)
+          ): Future[RlnRelayResult[WakuRlnRelay]] {.async.} =
+  ## Mounts the rln-relay protocol on the node.
+  ## The rln-relay protocol can be mounted in two modes: on-chain and off-chain.
+  ## Returns an error if the rln-relay protocol could not be mounted.
+  debug "rln-relay input validation passed"
+  try:
+    waku_rln_relay_mounting_duration_seconds.nanosecondTime:
+      let rlnRelay = await mount(conf,
+                                 registrationHandler)
+    return ok(rlnRelay)
+  except CatchableError as e:
+    return err(e.msg)
```

Comment:
These are the only lines changed from utils.nim, for any comments re: other code, I'll create a issue for it and address it in a follow up pr.